rootex/framework/components/visual/diffuse_visual_component.cpp

```diff
@@ -8,14 +8,18 @@
 
 #include "core/renderer/material.h"
 
-Component* DiffuseVisualComponent::Create(const LuaVariable& componentData)
+Component* DiffuseVisualComponent::Create(const JSON::json& componentData)
 {
-	ImageResourceFile* imageRes = ResourceLoader::CreateImageResourceFile(componentData["texturePath"].tostring());
+	ImageResourceFile* imageRes = ResourceLoader::CreateImageResourceFile(componentData["texturePath"]);
 	Ref<Texture> texture(new Texture(imageRes));
 
 	Ref<DiffuseMaterial> material(new DiffuseMaterial(texture));
 
-	DiffuseVisualComponent* diffuseComponent = new DiffuseVisualComponent(RenderPass::Global, material, ResourceLoader::CreateVisualModelResourceFile(componentData["resFile"].tostring()));
+	VisualModelResourceFile* vmr = ResourceLoader::CreateVisualModelResourceFile(componentData["resFile"]);
```

Comment:
Rename 'vmr'.

Reply:
vmr is an acronym for Visual Model Resource. Also this variable is local scope only and is basically a shorthand for VisualModelResourceFile

Reply:
Use of acronyms should be avoided, imo.

Reply:
Using acronyms is okay in smaller scopes. This variable lasts only for 7 lines of code. This is manageable